LiteCore/Query/QueryParser.cc

```diff
@@ -1268,6 +1271,30 @@ namespace litecore {
             return;
         }
 
+        // If the property in question is identified as an alias, emit that instead of
```

Comment:
In the existing code, if (db or join) aliases are in use, `_propertiesUseAliases` is true, and all properties must start with an alias. The code above from lines 1243-1251 identifies the alias and pulls it out of the path, and then line 1264 looks up the alias type, making your loop unnecessary.

The problem is that it would break compatibility to set `_propertiesUseAliases` when a result alias is defined, so we can't do that. So now we have a new state where the path might or might not begin with a result alias. Hm.

How about moving lines 1243-1251 up out of the `if` block, then looking up `alias` to see if it's a result alias and handling it if so. Otherwise proceed to the `if` block and so forth.

Reply:
The problem is that `_propertiesUseAliases` triggers logic in aliases that all apply to the way that *databases* are written.  What I want here is to change the way *results* are written.

Reply:
The path could either begin with a result alias or it could actually *be* a result alias (two cases to handle, the latter is not handled in that logic you mentioned since it doesn't contain a period)